lib/rtp_h264/nal_formats/fu.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Membrane.Element.RTP.H264.FU do
+  @moduledoc """
+  Module responsible for parsing H264 Fragmentation Unit.
+  """
+  use Bunch
+  alias Membrane.Element.RTP.H264.FU.Header
+  alias Membrane.Element.RTP.H264.{Depayloader, NAL}
+
+  defstruct [:last_seq_num, data: []]
+
+  @type t :: %__MODULE__{
+          data: [binary()],
+          last_seq_num: nil | Depayloader.sequence_number()
+        }
+
+  defguardp next?(last_seq_num, next_seq_num) when last_seq_num + 1 == next_seq_num
```

Comment:
Hey, @Hajto!
To be super specific, this is the place where we have something slightly unexpected and at the same time something that Gradualizer (due to its current limitations) warned about. `t.last_seq_num` is specced as `nil | Depayloader.sequence_number()`, while `next?` uses `+` (`last_seq_num + 1`) on it without first checking for `nil`. Since this is a guard, an error of this kind is silently ignored (which is the correct behaviour in your case) and matching falls through to another clause.
So it is a "bug" or a limitation, but rather in Gradualizer which can't yet figure all the parts which should be taken into account when checking guard expressions. I even checked if the guard can be more explicit by using `when last_seq_num != nil and last_seq_num + 1 == ...`, but [type refinement in Gradualizer is not sophisticated enough yet](https://github.com/josefs/Gradualizer/pull/109) for the tool to understand that `last_seq_num` just before `+` is not `nil | Depayloader.sequence_number()` anymore, but just `Depayloader.sequence_number()`.